stdlib/public/core/Integers.swift.gyb

```diff
@@ -218,40 +1134,344 @@ extension Arithmetic {
+  ///
+  /// In this example, `x` is initialized as an `Int8` instance by clamping
+  /// `500` to the range `-128...127`, and `y` is initialized as a `UInt`
+  /// instance by clamping `-500` to the range `0...UInt.max`.
+  ///
+  ///     let x = Int8(clamping: 500)
+  ///     // x == 127
+  ///     // x == Int8.max
+  ///
+  ///     let y = UInt(clamping: -500)
+  ///     // y == 0
+  ///
+  /// - Parameter source: An integer to convert to this type.
   init<T : BinaryInteger>(clamping source: T)
 
-  func word(at n: Int) -> UInt 
+  /// Returns the n-th word, counting from the least significant to most
+  /// significant, of this value's binary representation.
+  ///
+  /// The `word(at:)` method returns negative values in two's complement
+  /// representation, regardless of a type's underlying implementation. If `n`
+  /// is greater than the number of words in this value's current
+  /// representation, the result is `0` for positive numbers and `~0` for
+  /// negative numbers.
+  ///
+  /// - Parameter n: The word to return, counting from the least significant to
+  ///   most significant. `n` must be greater than or equal to zero.
+  /// - Returns: An word-sized, unsigned integer with the bit pattern of the
+  ///   n-th word of this value.
+  func word(at n: Int) -> UInt
+
+  /// The number of bits in the current binary representation of this value.
+  ///
+  /// This property is a constant for instances of fixed-width integer
+  /// types.
   var bitWidth : Int { get }
 
+  /// The number of bits required to represent this value in a signed type
+  /// using two's complement representation.
+  ///
+  /// The value of this property is always at least 1.
   var minimumSignedRepresentationBitWidth: Int { get }
 
 % for x in binaryArithmetic['BinaryInteger']:
   // defaulted using an in-place counterpart, but can be used as an
   // optimization hook
+${operationComment(x.operator)}
   func ${x.name}(${x.firstArg} rhs: Self) -> Self
 
   // implementation hook
+${mutatingOperationComment(x.operator)}
   mutating func ${x.mutatingName}(${x.firstArg} rhs: Self)
 % end
 
+  /// Returns the quotient and remainder of this value divided by the given
+  /// value.
+  ///
+  /// Use this method to calculate the quotient and remainder of a division at
+  /// the same time.
+  ///
+  ///     let x = 1_000_000
+  ///     let (q, r) = x.quotientAndRemainder(dividingBy: 933)
+  ///     // q == 1071
+  ///     // r == 757
+  ///
+  /// - Parameter rhs: The value to divide this value by.
+  /// - Returns: A tuple containing the quotient and remainder of this value
+  ///   divided by `rhs`.
   func quotientAndRemainder(dividingBy rhs: Self)
     -> (quotient: Self, remainder: Self)
 
-  /// Returns `-1` if the value of `self` is negative, `1` if it's positive,
-  /// `0` otherwise.
+  /// Returns `-1` if this value is negative and `1` if it's positive;
+  /// otherwise, `0`.
+  ///
+  /// - Returns: The sign of this number, expressed as an integer.
```

Comment:
'expressed as an integer' is not stressing the idea that the result has the same type as the value you call the method on. I think..